geoportailv3/static/js/draw/drawnfeaturesservice.js

```diff
@@ -1,26 +1,203 @@
+      this.appMymaps_.deleteFeature(feature);
+    }
+  }
+
+  this.encodeFeaturesInUrl_(this.features.getArray());
+};
+
+
+/**
+ * @param {ol.Feature} feature
+ */
+app.DrawnFeatures.prototype.add = function(feature) {
+  var features = this.features.getArray().slice();
+  features.push(feature);
+  this.saveFeature(feature);
+  this.encodeFeaturesInUrl_(features);
+};
+
+
+/**
+ * @param {Array.<ol.Feature>} features Features to encode in the URL.
+ * @private
+ */
+app.DrawnFeatures.prototype.encodeFeaturesInUrl_ = function(features) {
+  var featuresToEncode = features.filter(function(feature) {
+    return feature.get('__source__') != 'mymaps';
+  });
+
+  if (featuresToEncode.length > 0) {
+    this.ngeoLocation_.updateParams({
+      'features': this.fhFormat_.writeFeatures(featuresToEncode)
+    });
+  } else {
+    this.ngeoLocation_.deleteParam('features');
+  }
+};
+
+
 /**
- * @typedef {ol.Collection<ol.Feature>}
+ * @param {ol.Feature} feature the feature to save.
  */
-app.DrawnFeatures;
+app.DrawnFeatures.prototype.saveFeature = function(feature) {
+
+  if (this.appMymaps_.isEditable()) {
+    feature.set('__source__', 'mymaps');
+    this.saveFeatureInMymaps_(feature);
+  }
+  this.encodeFeaturesInUrl_(this.features.getArray());
+};
 
 
 /**
- * The selected features collection.
- * @type {app.DrawnFeatures}
- * @const
+ * Decode the features encoded in the URL and add them to the collection
+ * of drawn features.
+ */
+app.DrawnFeatures.prototype.drawFeaturesInUrl = function() {
+  var encodedFeatures = this.ngeoLocation_.getParam('features');
+  if (goog.isDef(encodedFeatures)) {
+    var remoteFeatures = this.fhFormat_.readFeatures(encodedFeatures);
+    goog.asserts.assert(!goog.isNull(remoteFeatures));
+    for (var i = 0; i < remoteFeatures.length; ++i) {
+      var feature = remoteFeatures[i];
+      var opacity = /** @type {string} */ (feature.get('opacity'));
+      feature.set('opacity', +opacity);
+      var stroke = /** @type {string} */ (feature.get('stroke'));
+      feature.set('stroke', +stroke);
+      var size = /** @type {string} */ (feature.get('size'));
+      feature.set('size', +size);
+      var angle = /** @type {string} */ (feature.get('angle'));
+      feature.set('angle', +angle);
+      var isLabel = /** @type {string} */ (feature.get('is_label'));
+      feature.set('is_label', isLabel === 'true');
+      feature.set('__editable__', true);
+      feature.set('__source__', 'url');
+      feature.setStyle(this.featureStyleFunction_);
+    }
+    this.features.extend(remoteFeatures);
+  }
+};
+
+
+/**
+ * @param {ol.Feature} feature Feature to encode in the URL.
  * @private
  */
-app.drawnFeatures_ = new ol.Collection();
+app.DrawnFeatures.prototype.saveFeatureInMymaps_ = function(feature) {
+  var currentFeature = feature;
+  if (this.appMymaps_.isEditable()) {
+    this.appMymaps_.saveFeature(feature)
+      .then(goog.bind(function(resp) {
+          var featureId = resp['id'];
```

Comment:
I would be best to have an externs file for the `mymaps` service responses. But I would not force you (or anyone else) to write one :)